plugin/mlsdbprovider.cpp

```diff
@@ -80,6 +81,11 @@ MlsdbProvider::MlsdbProvider(QObject *parent)
 
     staticProvider = this;
 
+    connect(&m_locationSettingsWatcher, SIGNAL(fileChanged(QString)),
+            this, SLOT(locationEnabledChanged()));
+    m_locationSettingsWatcher.addPath(LocationSettingsFile);
+    locationEnabledChanged();
```

Comment:
Maybe name as something like updateLocationEnabled could indicate the functionality better.